src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessModuleTests.cs

```diff
@@ -18,26 +19,30 @@ public void TestModulePropertiesExceptOnOSX()
             if (global::Interop.IsWindows)
                 fileName = string.Format("{0}.exe", CoreRunName);
 
-            // Ensure the process has loaded the modules.
-            Assert.True(SpinWait.SpinUntil(() =>
+            try
             {
-                if (_process.Modules.Count > 0)
-                    return true;
-                _process.Refresh();
-                return false;
-            }, WaitInMS));
+                // Ensure the process has loaded the modules.
+                Assert.True(SpinWait.SpinUntil(() =>
+                {
+                    if (_process.Modules.Count > 0)
+                        return true;
+                    _process.Refresh();
+                    return false;
+                }, WaitInMS));
 
-            ProcessModuleCollection moduleCollection = _process.Modules;
-            Assert.True(moduleCollection.Count > 0);
+                ProcessModuleCollection moduleCollection = _process.Modules;
+                Assert.True(moduleCollection.Count > 0);
 
-            ProcessModule coreRunModule = _process.MainModule;
-            Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
-            Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
-            Assert.Equal(fileName, coreRunModule.ModuleName);
-            Assert.True(coreRunModule.ModuleMemorySize > 0);
-            Assert.EndsWith(fileName, coreRunModule.FileName);
+                ProcessModule coreRunModule = _process.MainModule;
+                Assert.True(coreRunModule.BaseAddress.ToInt64() > 0);
+                Assert.True(coreRunModule.EntryPointAddress.ToInt64() >= 0);
+                Assert.Equal(fileName, coreRunModule.ModuleName);
+                Assert.True(coreRunModule.ModuleMemorySize > 0);
+                Assert.EndsWith(fileName, coreRunModule.FileName);
```

Comment:
It might be best to remove MainModule checks from here (MainModule test method would cover the FileName) and as @stephentoub suggested test the generic properties of all the modules here.